src/main/java/com/rapid7/client/dcerpc/mssamr/SAMPRDomainPasswordInfo.java

```diff
@@ -0,0 +1,166 @@
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_PASSWORD_STORE_CLEARTEXT | The server MUST store the cleartext password, not just the computed      |
+     * | 0x00000010                      | hashes.                                                                  |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     * | DOMAIN_REFUSE_PASSWORD_CHANGE   | Reserved. No effect on password policy.                                  |
+     * | 0x00000020                      |                                                                          |
+     * +---------------------------------+--------------------------------------------------------------------------+
+     */
+    private short minimumPasswordLength;
+    private short passwordHistoryLength;
+    private int passwordProperties;
+    private long maximumPasswordAge;
+    private long minimumPasswordAge;
+
+    public boolean isDomainPasswordComplex() {
+        return (this.passwordProperties & DOMAIN_PASSWORD_COMPLEX) != 0;
+    }
+
+    public boolean isDomainPasswordNoAnonChange() {
+        return (this.passwordProperties & DOMAIN_PASSWORD_NO_ANON_CHANGE) != 0;
+    }
+
+    public boolean isDomainPasswordStoredClearText() {
+        return (this.passwordProperties & DOMAIN_PASSWORD_STORE_CLEARTEXT) != 0;
+    }
+
+    public short getMinimumPasswordLength() {
+        return minimumPasswordLength;
+    }
+
+    public short getPasswordHistoryLength() {
+        return passwordHistoryLength;
+    }
+
+    public int getPasswordProperties() {
+        return passwordProperties;
+    }
+
+    public long getMaximumPasswordAge() {
+        return maximumPasswordAge;
+    }
+
+    public long getMinimumPasswordAge() {
+        return minimumPasswordAge;
+    }
+
+    @Override
+    public void unmarshalPreamble(PacketInput in) throws IOException {
+    }
+
+    @Override
+    public void unmarshalEntity(PacketInput in) throws IOException {
+        // Structure Alignment: 4
+        in.align(Alignment.FOUR);
+
+        this.minimumPasswordLength = in.readShort();
+        this.passwordHistoryLength = in.readShort();
+        this.passwordProperties = in.readInt();
+
+        // See <a href="http://technet.microsoft.com/en-us/library/cc753858(WS.10).aspx>Technet Article</a>
+        this.maximumPasswordAge = in.readLong();
+        this.minimumPasswordAge = in.readLong();
+    }
+
+    @Override
+    public void unmarshalDeferrals(PacketInput in) throws IOException {
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getMinimumPasswordLength(), getPasswordHistoryLength(), getPasswordProperties(),
+            getMaximumPasswordAge(), getMinimumPasswordAge(), isDomainPasswordComplex(), isDomainPasswordNoAnonChange(),
```

Comment:
You should not hash on `isDomainPasswordComplex` `isDomainPasswordNoAnonChange` or `isDomainPasswordStoredClearText`, these are only derivatives of `passwordProperties` and do not reflect a change in state.